### ISSUE TITLE
Long queued builds run against the wrong commit

### ISSUE THREAD

**Issue by QuLogic** [appveyor/ci#140]
Taking [this build](https://ci.appveyor.com/project/barsch/obspy/build/1.0.32) as an example, the commit that is tested in the first environment is different from the commit tested in the last.
 * The [first environment](https://ci.appveyor.com/project/barsch/obspy/build/1.0.32/job/v747e6mmrvinidi7) ran for 39 minutes and failed. (This is a side issue, but it seemed to do nothing, BTW.)
 * The [second environment](https://ci.appveyor.com/project/barsch/obspy/build/1.0.32/job/2cjr3288m3xgtjd1) is probably running against the correct commit (It's failing on `numpy/npy_common.h`, which is what I added in that commit.)
 * The [fourth environment](https://ci.appveyor.com/project/barsch/obspy/build/1.0.32/job/9jl1kjpb9t3avx8p) fails completely differently. That's because I've moved some scripts for the CI, and AppVeyor is attempting to build the latest commit, but using the original `appveyor.yml` that points to the old location.

**Comment by FeodorFitsner** [comment]
Yeah, this is because head commit of that PR is changing and command for retrieving virtual merge branch reflects that:

    git fetch -q origin +refs/pull/944/merge:

**Comment by am11** [comment]
@FeodorFitsner, will this issue be fixed or this is a behavior by design? I think this is a bug, as the only expected use-case is that the CI locks down the entire *jobs-set* to the commit under question instead of fetch-latest per commit for each job individually.

**Comment by FeodorFitsner** [comment]
How would we take exactly the same commit from that virtual merge branch? The problem of "pull" branch is that it's non-immutable. Every time you push some changes into PR it's being re-generated by GitHub and there is no warranty you'll find either head commit or merge commit. Thoughts?

**Comment by am11** [comment]
I think the key is the commit hash, that always remain in the history even if you `rebase` and `push -f` to overwrite the history. For instance, TravisCI does this:

```cmd
git clone --depth=50 --branch=684-proc git://github.com/am11/node-sass.git am11/node-sass
:: ...logs blah cloning ... boring...
cd am11/node-sass
git checkout -qf <SHA-1 commit hash here>
```

It keeps that hash and uses for each job in the given job-matrix.

**Comment by FeodorFitsner** [comment]
Yeah, and we *do* checkout particular commit for regular builds too. However, for PR builds it takes HEAD commit of pull branch, for instance: https://travis-ci.org/gruntjs/grunt/jobs/51794452

**Comment by am11** [comment]
This is true. The thing is on TravisCI the job builds often start in parallel (even for OSS repos where "Concurrent jobs" is set to one in Settings), therefore it is significantly ignorable. While with AppVeryor, the gap is quite high and yet the magnitude of this issue magnifies. 

On a separate note: currently their status page is saying they are having some issues: http://traviscistatus.com/, would be nice to have similar status page on AppVeyor.
